django_cloud_tasks/models.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+from django.db import transaction
+from django.apps import apps
+
+from datetime import datetime
+from django.db import models
+from django_cloud_tasks import tasks
+from django_cloud_tasks.errors import PipelineError
+
+class Pipeline(models.Model):
+    name = models.CharField(max_length=50)
+
+    def start(self):
+        routines = self.routines.filter(next_routines__id__isnull=True)
+        if not routines.exists():
+            raise PipelineError("Not enought routines to start")
+
+        if routines.count() > 1:
+            raise PipelineError("Too many routines to start at once")
+
+        routines.first().enqueue()
+
+    def revert(self):
+        for routine in self.routines.all():
+            routine.revert()
```

Comment:
We should not revet all at once, it'll be easier to do the same strategy it's done on start: the pipeline chose the starting points, and these points triggers the next points.

On start: pipeline locates root routines, and triggers all of them. Each finished routine start the their next routines.
On revert: pipeline locates leaf routines, and triggers all of them. Each reverted routine trigger revert their previous routines.